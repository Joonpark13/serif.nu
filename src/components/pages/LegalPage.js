import React from 'react';
import { Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import { pageContainer, pageBody } from './common/styles';
import PageTitle from './common/PageTitle';

const useStyles = makeStyles({
  pageContainer,
  pageBody,
  section: {
    margin: '30px 0',
  },
});

export default function LegalPage() {
  const classes = useStyles();
  return (
    <div className={classes.pageContainer}>
      <PageTitle title="Serif.nu - Legal" />

      <div className={classes.pageBody}>
        <Typography variant="body1">
          &copy; 2017
          {' '}
          <a href="http://joonpark.me/">Joon Park</a>
        </Typography>
        <Typography variant="body1">
          Serif.nu is not created by Northwestern University.
        </Typography>

        <div className={classes.section}>
          <Typography variant="h4">
            Terms of Service
          </Typography>

          <ol>
            <li>
              <Typography variant="h6">Terms</Typography>
              <Typography variant="body1">
                By accessing this web site, you are agreeing to be bound by these web site Terms of
                Service, all applicable laws and regulations, and agree that you are responsible for
                compliance with any applicable local laws. If you do not agree with any of these
                terms, you are prohibited from using or accessing this web site. The materials
                contained in this web site are protected by applicable copyright and trade mark law.
              </Typography>
            </li>
            <li>
              <Typography variant="h6">Disclaimer</Typography>
              <Typography variant="body1">
                The materials on Serif.nu are provided &quot;as is.&quot; Serif.nu makes no
                warranties, expressed or implied, and hereby disclaims and negates all other
                warranties, including without limiation, implied warranties or conditions of
                merchantability, fitness for a particular purpose, or non-infringement of
                intellectual property or other violation of rights. Further, Serif.nu does not
                warrant or make any representations concerning the accuracy, likely results, or
                reliablility of the use of the materials on the web site or otherwise relating to
                such materials or on any web sites linked to this web site.
              </Typography>
            </li>
            <li>
              <Typography variant="h6">Errata</Typography>
              <Typography variant="body1">
                The materials appearing on Serif.nu&apos;s web site could include technical,
                typographical, or photographic errors. Serif.nu does not warrant that any of the
                materials on its web site are accurate, complete, or current. Serif.nu may make
                changes to the materials contained on its web site at any time without notice.
              </Typography>
            </li>
            <li>
              <Typography variant="h6">Limitations</Typography>
              <Typography variant="body1">
                In no ever shall Serif.nu or its creators be liable for any damages (including,
                without limitation, damanges for loss of data or profit, or due to business
                interruption) arising out of the use or inability to use the materials on Serif.nu,
                even if Serif.nu or Serif.nu&apos;s creators have been notified orally or in writing
                of the possibility of such damage.
              </Typography>
            </li>
            <li>
              <Typography variant="h6">Links</Typography>
              <Typography variant="body1">
                Serif.nu has not reviewed all of the sites linked to its Internet web site and is
                not responsible for the contents of any such linked site. The inclusion of any link
                does not imply endorsement by Serif.nu of the site. Use of any such linked web site
                is at the user&apos;s own risk.
              </Typography>
            </li>
            <li>
              <Typography variant="h6">Terms of Service Modifications</Typography>
              <Typography variant="body1">
                Serif.nu may revise these Terms of Service at any time without notice. By using this
                web site you are agreeing to be bound by the current version of the Terms of Service
                at the time.
              </Typography>
            </li>
            <li>
              <Typography variant="h6">Governing Law</Typography>
              <Typography variant="body1">
                Any claim relating to Serif.nu shall be governed by the laws of the State of
                Illinois without regard to its conflict of law provisions.
              </Typography>
            </li>
          </ol>
        </div>

        <div className={classes.section}>
          <Typography variant="h4" gutterBottom>
            Privacy Policy
          </Typography>

          <Typography variant="body1">
            Your privacy is very important to us. Accordingly, we have developed this Policy in
            order for you to understand how we collect, use, communicate and disclose and make use
            of personal information. The following outlines our privacy policy.
          </Typography>

          <Typography variant="body1">
            <ul>
              <li>
                We will collect and use of personal information solely with the objective of
                bettering the user&apos;s experience or for other compatible purposes.
              </li>
              <li>
                We will only retain personal information as long as necessary for the fulfillment of
                those purposes.
              </li>
              <li>
                We will collect personal information by lawful and fair means and, where
                appropriate, with the knowledge or consent of the individual concerned.
              </li>
              <li>
                We will attempt to protect personal information by reasonable security safeguards
                against loss or theft, as well as unauthorized access, disclosure, copying, use or
                modification.
              </li>
              <li>
                We will make readily available to users information about our policies and practices
                relating to the management of personal information.
              </li>
              <li>
                We may use analytics software, including but not limited to, Google Analytics, etc.
                Such data collected will be used solely for the purpose of improving the user
                experience.
              </li>
              <li>
                We use local storage technologies, including Cookies, localStorage, etc. to enhance
                your experience.
              </li>
            </ul>
          </Typography>
        </div>
      </div>
    </div>
  );
}
