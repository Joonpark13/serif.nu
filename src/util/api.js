/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
