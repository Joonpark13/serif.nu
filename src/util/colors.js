export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
