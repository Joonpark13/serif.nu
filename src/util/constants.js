/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
