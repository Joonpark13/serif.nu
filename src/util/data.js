import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
