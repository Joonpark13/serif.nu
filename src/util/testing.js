/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
