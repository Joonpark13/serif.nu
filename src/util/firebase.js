import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
