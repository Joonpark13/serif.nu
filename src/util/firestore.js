import firebase from 'firebase/app';
import 'firebase/firestore';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export default firebase.firestore();
