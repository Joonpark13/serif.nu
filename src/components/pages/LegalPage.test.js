import { wrapperCreator } from 'util/testing';
import { UnstyledLegalPage, styles } from './LegalPage';

describe('LegalPage', () => {
  const getWrapper = wrapperCreator(UnstyledLegalPage, undefined, styles);

  it('should render correctly', () => {
    const wrapper = getWrapper();

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
