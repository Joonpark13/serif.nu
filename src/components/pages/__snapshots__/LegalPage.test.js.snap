// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegalPage should render correctly 1`] = `
<div
  className="makeStyles-pageContainer-1"
>
  <PageTitle
    title="Serif.nu - Legal"
  />
  <div
    className="makeStyles-pageBody-2"
  >
    <ForwardRef(WithStyles)
      variant="body1"
    >
      Â© 2017
       
      <a
        href="http://joonpark.me/"
      >
        Joon Park
      </a>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      variant="body1"
    >
      Serif.nu is not created by Northwestern University.
    </ForwardRef(WithStyles)>
    <div
      className="makeStyles-section-3"
    >
      <ForwardRef(WithStyles)
        variant="h4"
      >
        Terms of Service
      </ForwardRef(WithStyles)>
      <ol>
        <li>
          <ForwardRef(WithStyles)
            variant="h6"
          >
            Terms
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            variant="body1"
          >
            By accessing this web site, you are agreeing to be bound by these web site Terms of Service, all applicable laws and regulations, and agree that you are responsible for compliance with any applicable local laws. If you do not agree with any of these terms, you are prohibited from using or accessing this web site. The materials contained in this web site are protected by applicable copyright and trade mark law.
          </ForwardRef(WithStyles)>
        </li>
        <li>
          <ForwardRef(WithStyles)
            variant="h6"
          >
            Disclaimer
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            variant="body1"
          >
            The materials on Serif.nu are provided "as is." Serif.nu makes no warranties, expressed or implied, and hereby disclaims and negates all other warranties, including without limiation, implied warranties or conditions of merchantability, fitness for a particular purpose, or non-infringement of intellectual property or other violation of rights. Further, Serif.nu does not warrant or make any representations concerning the accuracy, likely results, or reliablility of the use of the materials on the web site or otherwise relating to such materials or on any web sites linked to this web site.
          </ForwardRef(WithStyles)>
        </li>
        <li>
          <ForwardRef(WithStyles)
            variant="h6"
          >
            Errata
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            variant="body1"
          >
            The materials appearing on Serif.nu's web site could include technical, typographical, or photographic errors. Serif.nu does not warrant that any of the materials on its web site are accurate, complete, or current. Serif.nu may make changes to the materials contained on its web site at any time without notice.
          </ForwardRef(WithStyles)>
        </li>
        <li>
          <ForwardRef(WithStyles)
            variant="h6"
          >
            Limitations
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            variant="body1"
          >
            In no ever shall Serif.nu or its creators be liable for any damages (including, without limitation, damanges for loss of data or profit, or due to business interruption) arising out of the use or inability to use the materials on Serif.nu, even if Serif.nu or Serif.nu's creators have been notified orally or in writing of the possibility of such damage.
          </ForwardRef(WithStyles)>
        </li>
        <li>
          <ForwardRef(WithStyles)
            variant="h6"
          >
            Links
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            variant="body1"
          >
            Serif.nu has not reviewed all of the sites linked to its Internet web site and is not responsible for the contents of any such linked site. The inclusion of any link does not imply endorsement by Serif.nu of the site. Use of any such linked web site is at the user's own risk.
          </ForwardRef(WithStyles)>
        </li>
        <li>
          <ForwardRef(WithStyles)
            variant="h6"
          >
            Terms of Service Modifications
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            variant="body1"
          >
            Serif.nu may revise these Terms of Service at any time without notice. By using this web site you are agreeing to be bound by the current version of the Terms of Service at the time.
          </ForwardRef(WithStyles)>
        </li>
        <li>
          <ForwardRef(WithStyles)
            variant="h6"
          >
            Governing Law
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            variant="body1"
          >
            Any claim relating to Serif.nu shall be governed by the laws of the State of Illinois without regard to its conflict of law provisions.
          </ForwardRef(WithStyles)>
        </li>
      </ol>
    </div>
    <div
      className="makeStyles-section-3"
    >
      <ForwardRef(WithStyles)
        gutterBottom={true}
        variant="h4"
      >
        Privacy Policy
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        variant="body1"
      >
        Your privacy is very important to us. Accordingly, we have developed this Policy in order for you to understand how we collect, use, communicate and disclose and make use of personal information. The following outlines our privacy policy.
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        variant="body1"
      >
        <ul>
          <li>
            We will collect and use of personal information solely with the objective of bettering the user's experience or for other compatible purposes.
          </li>
          <li>
            We will only retain personal information as long as necessary for the fulfillment of those purposes.
          </li>
          <li>
            We will collect personal information by lawful and fair means and, where appropriate, with the knowledge or consent of the individual concerned.
          </li>
          <li>
            We will attempt to protect personal information by reasonable security safeguards against loss or theft, as well as unauthorized access, disclosure, copying, use or modification.
          </li>
          <li>
            We will make readily available to users information about our policies and practices relating to the management of personal information.
          </li>
          <li>
            We may use analytics software, including but not limited to, Google Analytics, etc. Such data collected will be used solely for the purpose of improving the user experience.
          </li>
          <li>
            We use local storage technologies, including Cookies, localStorage, etc. to enhance your experience.
          </li>
        </ul>
      </ForwardRef(WithStyles)>
    </div>
  </div>
</div>
`;
